omnipaxos_core/src/omni_paxos.rs

```diff
@@ -224,6 +237,18 @@ where
         self.seq_paxos.reconnected(pid)
     }
 
+    /// Drives the election process (see `election_timeout()`) every `election_tick_timeout`
+    /// ticks. Also drives the detection and resending of dropped omnipaxos messages every `resend_message_tick_timeout` ticks.
+    pub fn tick(&mut self) {
+        self.tick += 1;
+        if self.tick % self.election_tick_timeout == 0 {
+            self.election_timeout();
+        }
+        if self.tick % self.resend_message_tick_timeout == 0 {
+            self.seq_paxos.resend_message_timeout();
```

Comment:
I initially reset the tick to 0 when the tick equaled the product of the timeouts, but then I realized if you tick once every millisecond you will have enough ticks for 584942417 years. But a `Logical Clock` struct could be nicer. We have a similar problem with the sequence numbers which could also overflow.

Reply:
There is no reason to leave the possibility of overflow if there is a way to prevent it. With sequence numbers, I don't think there is a better way, and it also gets reset per leader change. It would be amazing if OmniPaxos didn't even change leader for that many years :D

Reply:
We could use the overflow as our mechanism for resetting the ticks. I think unsigned int overflow is defined behavior in Rust? Then every 1/2 billion-ish years your election timeout and resending messages may get triggered earlier than expected, but nothing breaks.

This can solve the sequence number overflow problem aswell. TCP sequence numbers do this [wrap-around](https://www.geeksforgeeks.org/wrap-around-concept-and-tcp-sequence-number/) actually.

Reply:
No, just reset the logical clock to 0 when it reaches its timeout. If you want we can fix the wrap-around for message seq. numbers in a future PR.

Reply:
Using custom handling for overflow is an overkill for this case and just makes the code less intuitive.